### ISSUE TITLE
Training Concatenated Pre-Trained Models with Concatenated Data Generators

### ISSUE THREAD

**Issue by emresahinn19** [keras-team/keras#18922]
I have combined two pre-trained models, ResNet50 and ResNet50V2, and the concatenated data generator and models appear to be set up correctly as I can initiate the training process. However, during training, I observe unusually high loss values while accuracy seems reasonable. After waiting for 20 minutes, the loss has reached approximately 2.2, and accuracy is around 0.88. The challenge is that the model did not progress to the second epoch within this time frame. I am struggling to identify the underlying issue. After 20 minutes true_positive number was around 24500. I don't think that I have so much train images even with data augmentation. 

Edit: After setting steps_per_epoch = 7, it is still stucking on epoch 1 and nothing changes within time. And there is no validation results. 

Both datasets contain 224 images for training, and they are of different types. 

Data generator code:
 

```python
train_datagen = ImageDataGenerator(
                                  preprocessing_function=keras.applications.resnet50.preprocess_input,
                                   validation_split=0.20,
                                   vertical_flip=True, 
                                   horizontal_flip=True,
                                   zoom_range=0.4


                                   
                                  )



def generate_generator_multiple(generator,dir1, dir2, batch_size, img_height,img_width):
    genX1 = generator.flow_from_directory(dir1,
                                          target_size = (img_height,img_width),
                                          class_mode = 'binary',
                                          batch_size = batch_size,
                                          shuffle=True
                                          )

    genX2 = generator.flow_from_directory(dir2,
                                          target_size = (img_height,img_width),
                                          class_mode = 'binary',
                                          batch_size = batch_size,
                                          shuffle=True
                                          )
    while True:

            X1i = genX1.next()
            X2i = genX2.next()
            yield [X1i[0], X2i[0]], X1i[1]  




train_dir_1 = '/content/2/train'
train_dir_2 = '/content/3/train'
batch_size = 32
img_height = 224
img_width = 224

train_generator=generate_generator_multiple(generator=train_datagen,
                                           dir1=train_dir_1,
                                           dir2=train_dir_2,
                                           batch_size=batch_size,
                                           img_height=img_height,
                                           img_width=img_height)

val_generator=generate_generator_multiple(generator = train_datagen,
                                          dir1=train_dir_1,
                                          dir2=train_dir_2,
                                          batch_size=batch_size,
                                          img_height=img_height,
                                          img_width=img_height)
X = next(train_generator)
``` 
Model:
```python
first_model = keras.applications.ResNet50(
     weights='imagenet',
     input_shape=(224, 224, 3),
     include_top=False
     )


first_model.trainable = False
first_inputs = keras.Input(shape=(224, 224, 3))

x = first_model(first_inputs, training=False)

x = keras.layers.GlobalAveragePooling2D()(x)


#SECOND MODEL#
second_model = keras.applications.ResNet50V2(
     weights='imagenet',
     input_shape=(224, 224, 3),
     include_top=False
     )


second_model.trainable = False
second_inputs = keras.Input(shape=(224, 224, 3))

x1 = second_model(second_inputs, training=False)

x1 = keras.layers.GlobalAveragePooling2D()(x1)


### CONCATENATE TWO MODEL ###

concatenate_model = keras.layers.concatenate([x, x1])

outputs = Dense( units = 128, activation  = 'relu')(concatenate_model)

concantenate_model = keras.layers.Dropout(0.25)(concatenate_model)

outputs = Dense( units = 128, activation  = 'relu')(concatenate_model)

concantenate_model = keras.layers.Dropout(0.25)(concatenate_model)

outputs = Dense( units = 1, activation  = 'sigmoid')(concatenate_model)
combined_model = keras.Model(inputs=[first_inputs, second_inputs], outputs=outputs) #combined_model olacak.
keras.utils.plot_model(combined_model, show_shapes=True)
``` 
Model compile:
```python
combined_model.compile(
    optimizer=Adam(learning_rate=1e-4),
    loss='binary_crossentropy', #binary yap
    metrics=['accuracy',tf.keras.metrics.TruePositives(thresholds=0.5),
             tf.keras.metrics.TrueNegatives(thresholds=0.5),
             tf.keras.metrics.FalseNegatives(thresholds=0.5),
             tf.keras.metrics.FalsePositives(thresholds=0.5)
           ]
)
early_stopping = tf.keras.callbacks.EarlyStopping(
    patience=15,
    min_delta=0.001,
    restore_best_weights=True,
)

callbacks = [early_stopping]
history = combined_model.fit(
    train_generator,
    shuffle = True,
    steps_per_epoch = 7,
    validation_data=val_generator,
    batch_size=32,
    callbacks = callbacks,
    epochs=10,
    verbose = 1
)
```

**Comment by SuryanarayanaY** [comment]
Hi @emresahinn19 ,

Could you please submit the code in the form of a colab gist as the code seems not organized properly?

**Comment by emresahinn19** [comment]
> Hi @emresahinn19 ,
> 
> Could you please submit the code in the form of a colab gist as the code seems not organized properly?

Is it okey now?

**Comment by emresahinn19** [comment]
> Hi @emresahinn19 ,
> 
> Could you please submit the code in the form of a colab gist as the code seems not organized properly?

I created Github gist.

**Comment by SuryanarayanaY** [comment]
Hi @emresahinn19 ,

The link seems wrong or not working. Could you please check and provide exact one?

**Comment by emresahinn19** [comment]
> Hi @emresahinn19 ,
> 
> The link seems wrong or not working. Could you please check and provide exact one?
Check again. Now seems fine.

**Comment by emresahinn19** [comment]
By correctly setting the validation steps, the issue has been resolved. This topic can be closed.

**Comment by SuryanarayanaY** [comment]
Hi @emresahinn19 ,

Thanks for confirmation.Closing the issue as resolved now. Please feel free to reopen this or file a new issue if this is still an issue.
